app/models/rubygem.rb

```diff
@@ -313,32 +351,30 @@ def compact_index_info
     end
   end
 
-  private
+  private_class_method def self.versions_after(date)
+    created_gems = Rubygem.joins(:versions)
+      .select("name, versions.created_at as date, number, platform, info_checksum")
+      .where("versions.created_at > ?", date)
+      .order("versions.created_at, versions.number, platform")
 
-  def ensure_name_format
-    if name.class != String
-      errors.add :name, "must be a String"
-    elsif name !~ /[a-zA-Z]+/
-      errors.add :name, "must include at least one letter"
-    elsif name !~ NAME_PATTERN
-      errors.add :name, "can only include letters, numbers, dashes, and underscores"
-    end
-  end
+    yanked_gems = Rubygem.joins(:versions)
+      .select("name, versions.yanked_at as date, '-'||number, platform, info_checksum")
+      .where("indexed is false and versions.created_at > ?", date)
+      .order("versions.created_at, versions.number, platform")
 
-  def needs_name_validation?
-    new_record? || name_changed?
-  end
+    # Arel don't support yet order on unions :(
+    # https://github.com/rails/arel/issues/98
+    sql = "(#{created_gems.to_sql}) UNION (#{yanked_gems.to_sql}) ORDER BY date"
+    gems = Rubygem.find_by_sql(sql)
```

Comment:
Using raw sql here *drops* response time of `versions` from: 9072-9965ms to 2766-3182ms (min-max over 10 requests). I will make this change after this PR gets merged.

ENV:
config.cache_classes = true
config.consider_all_requests_local       = false
config.action_controller.perform_caching = false
versions file: one on bundler-api (created_at: 2016-01-22T16:03:56-06:00)